account_invoice_edifact/models/account_move.py

```diff
@@ -0,0 +1,214 @@
+# Copyright 2023 Camtocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def edifact_export_invoice(self):
+        self.ensure_one()
+        edifact_model = self.env["base.edifact"]
+        lines = []
+        interchange = self._edifact_export_invoice_interchange()
+
+        header = self._edifact_export_invoice_header()
+        product = self._edifact_export_invoice_product()
+        summary = self._edifact_export_invoice_summary()
+        lines += header + product + summary
+        for segment in lines:
+            interchange.add_segment(edifact_model.create_segment(*segment))
+        return interchange.serialize()
+
+    def _edifact_export_invoice_interchange(self):
+        self.ensure_one()
+        id_number = self.env["res.partner.id_number"]
+        sender = id_number.search(
+            [("partner_id", "=", self.invoice_user_id.partner_id.id)]
+        )
+        recipient = id_number.search([("partner_id", "=", self.partner_id.id)])
+        if not sender or not recipient:
+            raise UserError(_("Partner is not allowed to use the feature."))
+        sender_edifact = [sender.name, "14"]
+        recipient_edifact = [recipient.name, "14"]
+        syntax_identifier = ["UNOC", "3"]
+
+        return self.env["base.edifact"].create_interchange(
+            sender_edifact, recipient_edifact, self.id, syntax_identifier
+        )
+
+    def _edifact_export_invoice_header(self):
+        self.ensure_one()
+        source_orders = self.line_ids.sale_line_ids.order_id
+        today = datetime.now().date().strftime("%Y%m%d")
+        id_number = self.env["res.partner.id_number"]
+        buyer = self.partner_id
+
+        seller = self.invoice_user_id.partner_id
+        seller_id_number = id_number.search([("partner_id", "=", seller.id)])
+
+        shipper = self.partner_shipping_id
+        shipper_id_number = id_number.search([("partner_id", "=", shipper.id)])
+        return [
+            ("UNH", "1", ["INVOIC", "D", "96A", "UN", "EAN008"]),
+            ("BGM", ["380", "", "", "Invoice"], self.payment_reference, "9"),
+            ("DTM", ["35", "20230928", "102"]),  # TODO: Change datetime
```

Comment:
```suggestion
            # 35: Delivery date/time, actual
            ("DTM", ["35", "20230928", "102"]),  # TODO: Change datetime
```

Reply:
if any of the related picking has been done already, take `min(date_done)`
if not, don't put this part at all